### DISCUSSION TITLE
Doom Emacs

### DISCUSSION THREAD

**Discussion by dboneham** [emacs-eaf/emacs-application-framework#475] [General]
I couldn't see any notes on how to install into Doomemacs - which I use.

I managed to do this in two basic steps, as follows:

1. Insert the following text into packages.el (one of the three main config files in Doomemacs).
(package! emacs-application-framework
  :recipe (:host github :repo "manateelazycat/emacs-application-framework"
           :files ("eaf.el" "src/lisp/*.el")))
2. Doing a git clone of the repo into ~/.emacs.d/.local/straight/build-27.1/emacs-application framework/
I had to move eaf.el and emacs-application-framework-autoloads.el to empty the directory before cloning. Then moved the file back afterwards.

There is probably a better way of installing eaf into Doomcasts (without disrupting its processes).

Here is a copy of my eaf config within config.el:
(setq browse-url-browser-function 'eaf-open-browser)
(defalias 'browse-web #'eaf-open-browser)
(setq eaf-browser-enable-adblocker "true")
(setq eaf-browser-continue-where-left-off t)
(setq eaf-browser-default-search-engine "duckduckgo")
(setq eaf-browse-blank-page-url "https://duckduckgo.com")
(setq eaf-browser-default-zoom "3")
(require 'eaf-org)
(defun eaf-org-open-file (file &optional link)
  "An wrapper function on `eaf-open'."
  (eaf-open file))
;; use `emacs-application-framework' to open PDF file: link
(add-to-list 'org-file-apps '("\\.pdf\\'" . eaf-org-open-file))

I am not currently using the evil set-up within eaf for fear of disrupting Doom's own highly developed Evil settings.

Interested in ay tips in relation to using eaf within Doomemacs

**Comment by MatthewZMD** [comment]
Since Github discussion is still a fairly new feature, I dunno whether there are a lot of people looking here and you'll get a solution any time soon. Maybe you should ask on reddit or create an issue instead.

**Comment by dboneham** [comment]
That's OK. I'm not looking for a solution, as such. I just thought 
any Doomemacs users interested in installing EAF might wish to 
learn how I did it.

Best wishes,

David Boneham

Matthew Zeng <notifications@github.com> writes:

> Since Github discussion is still a fairly new feature, I dunno 
> whether there are a lot of people looking here and you'll get a 
> solution any time soon. Maybe you should ask on reddit or create 
> an issue instead.



--
---
David Boneham

**Reply by MatthewZMD** [reply]
Oh that's good to know! Sorry I made the wrong assumption, as I'm not a Doom Emacs user myself and didn't pay too much attention on the context.

**Comment by loyalpartner** [comment]
@dboneham 
```
(when (package! eaf :recipe (:host github
                             :repo "manateelazycat/emacs-application-framework"
                             :files ("*.el" "*.py" "app" "core")
                             :build (:not compile)))

  (package! ctable :recipe (:host github :repo "kiwanami/emacs-ctable"))
  (package! deferred :recipe (:host github :repo "kiwanami/emacs-deferred"))
  (package! epc :recipe (:host github :repo "kiwanami/emacs-epc")))

(use-package! eaf
  :commands (eaf-open-browser eaf-open find-file)
  :config
  (use-package! ctable)
  (use-package! deferred)
  (use-package! epc))
```

**Reply by thyecust** [reply]
I've add this snippet to `$DOOMDIR/pacakges.el` and run `doom sync`, but it seems that EAF still can't work. For example after running `M-x eaf-open-browser` and typing a URL, I can only get a blank window, rather than a browser. I tried to run the `install.sh`. Is there something wrong with what I'm doing?

**Reply by colawithsauce** [reply]
That's because doom would build the packages and place them into the 'build' directory. Running `eaf-install-dependencies` is to running `./install-eaf.sh` in the `build` directory. However, when running `eaf-open-browser`, the program would find these dependencies in `repo` directory.

So the solution is running `./install-eaf.sh` in your `repo` directory or just manage emacs-application-framework manually, instead of calling `eaf-install-dependencies`

#722

**Reply by thyecust** [reply]
> That's because doom would build the packages and place them into the 'build' directory. Running `eaf-install-dependencies` is to running `./install-eaf.sh` in the `build` directory. However, when running `eaf-open-browser`, the program would find these dependencies in `repo` directory.
> 
> So the solution is running `./install-eaf.sh` in your `repo` directory or just manage emacs-application-framework manually, instead of calling `eaf-install-dependencies`
> 
> #722

Now I turn to naive emacs and EAF without doom. But thanks for your solution.

**Comment by stan-ely** [comment]
Looks like doom Emacs uses straight for package management.

I've got a working straight configuration for eaf which does all the cloning and building.

[Link](https://github.com/emacs-eaf/emacs-application-framework/discussions/799#discussioncomment-2857758) to the conf.

**Comment by osvarcha** [comment]
I've had a headache to match EAF with Straight.el so go for the following setup in $ DOOMDIR config.el or config.org
```emacs-lisp
(use-package! eaf
  :load-path "~/.elisp/emacs-application-framework"
  :init
  :custom
  (eaf-browser-continue-where-left-off t) 
  (eaf-browser-enable-adblocker t)
  (browse-url-browser-function 'eaf-open-browser) ;; Make EAF Browser my default browser
  :config
  (defalias 'browse-web #'eaf-open-browser)

  (require 'eaf-file-manager)
  (require 'eaf-music-player)
  (require 'eaf-image-viewer)
  (require 'eaf-camera)
  (require 'eaf-demo)
  (require 'eaf-airshare)
  (require 'eaf-terminal)
  (require 'eaf-markdown-previewer)
  (require 'eaf-video-player)
  (require 'eaf-vue-demo)
  (require 'eaf-file-sender)
  (require 'eaf-pdf-viewer)
  (require 'eaf-mindmap)
  (require 'eaf-netease-cloud-music)
  (require 'eaf-jupyter)
  (require 'eaf-org-previewer)
  (require 'eaf-system-monitor)
  (require 'eaf-rss-reader)
  (require 'eaf-file-browser)
  (require 'eaf-browser)
  (require 'eaf-org)
  (require 'eaf-mail)
  (require 'eaf-git)
  (when (display-graphic-p)
    (require 'eaf-all-the-icons))

  (require 'eaf-evil)
  (define-key key-translation-map (kbd "SPC")
    (lambda (prompt)
      (if (derived-mode-p 'eaf-mode)
          (pcase eaf--buffer-app-name
            ("browser" (if  (string= (eaf-call-sync "call_function" eaf--buffer-id "is_focus") "True")
                           (kbd "SPC")
                         (kbd eaf-evil-leader-key)))
            ("pdf-viewer" (kbd eaf-evil-leader-key))
            ("image-viewer" (kbd eaf-evil-leader-key))
            (_  (kbd "SPC")))
        (kbd "SPC")))))
```

It is the best way to get EAF at DoomEmacs, I will be attentive if someone manages to install it in a better way.
